patches/chrome-browser-sync-chrome_sync_client.cc.patch

```diff
@@ -0,0 +1,22 @@
+diff --git a/chrome/browser/sync/chrome_sync_client.cc b/chrome/browser/sync/chrome_sync_client.cc
+index b949c22d19dc7ac956ee283645c0a34f78b392be..a8e158c6ed5aeda65e68d7a5ce0a234769864607 100644
+--- a/chrome/browser/sync/chrome_sync_client.cc
++++ b/chrome/browser/sync/chrome_sync_client.cc
+@@ -214,7 +214,7 @@ bool IsAppSyncEnabled(Profile* profile) {
+ 
+ }  // namespace
+ 
+-ChromeSyncClient::ChromeSyncClient(Profile* profile)
++ChromeSyncClient::ChromeSyncClient(Profile* profile, bool is_rewards_chrome_sync_client)
+     : profile_(profile), extensions_activity_monitor_(profile) {
+   DCHECK_CURRENTLY_ON(BrowserThread::UI);
+ 
+@@ -249,7 +249,7 @@ ChromeSyncClient::ChromeSyncClient(Profile* profile)
+       BookmarkSyncServiceFactory::GetForProfile(profile_));
+ 
+ #if BUILDFLAG(IS_ANDROID)
+-  trusted_vault_client_ = std::make_unique<TrustedVaultClientAndroid>();
++  if (!is_rewards_chrome_sync_client) trusted_vault_client_ = std::make_unique<TrustedVaultClientAndroid>();
```

Comment:
these changes aren't necessary, they can be handled in the subclass. If you don't need TrustedVaultClientAndroid (are you sure about that?) then you can return nullptr in `GetTrustedVaultClient`

Reply:
We do need `if (!is_rewards_chrome_sync_client) trusted_vault_client_ = std::make_unique<TrustedVaultClientAndroid>();`. Here's what we get on constructing `TrustedVaultClientAndroid` (the second time) otherwise:
```
Stack Trace:
  RELADDR   FUNCTION                                                                          FILE:LINE
  00106e47  base::debug::StackTrace::StackTrace(unsigned int)                                 ../../base/debug/stack_trace.cc:221:12
  00117ed1  logging::LogMessage::~LogMessage()                                                ../../../src/base/logging.cc:600:29
  001b09c1  base::android::CheckException(_JNIEnv*)                                           ../../base/android/jni_android.cc:306:3
  v------>  jni_generator::CheckException(_JNIEnv*)                                           ../../base/android/jni_generator/jni_generator_helper.h:53:3
  00ebfeef  jni_generator::JniJavaCallContextChecked::~JniJavaCallContextChecked()            ../../base/android/jni_generator/jni_generator_helper.h:118:5
  v------>  Java_TrustedVaultClient_registerNative(_JNIEnv*, long long)                       gen/chrome/android/chrome_jni_headers/TrustedVaultClient_jni.h:146:1
  01d48809  TrustedVaultClientAndroid::TrustedVaultClientAndroid()                            ../../chrome/browser/sync/trusted_vault_client_android.cc:49:3
  01c64663  browser_sync::ChromeSyncClient::ChromeSyncClient(Profile*)                        ../../../src/chrome/browser/sync/chrome_sync_client.cc:254:27
  01c671bb  (anonymous namespace)::BuildSyncService(content::BrowserContext*)                 ../../../src/chrome/browser/sync/sync_service_factory.cc:91:7
  00008ec5  KeyedServiceFactory::GetServiceForContext(void*, bool)                            ../../components/keyed_service/core/keyed_service_factory.cc:80:15
  01c522d5  SharingServiceFactory::BuildServiceInstanceFor(content::BrowserContext*) const    ../../chrome/browser/sharing/sharing_service_factory.cc:92:7
  00008ec5  KeyedServiceFactory::GetServiceForContext(void*, bool)                            ../../components/keyed_service/core/keyed_service_factory.cc:80:15
  01c22d6d  ProfileImpl::DoFinalInit(Profile::CreateMode)                                     ../../../src/chrome/browser/profiles/profile_impl.cc:817:3
  01c2374d  ProfileImpl::OnLocaleReady(Profile::CreateMode)                                   ../../../src/chrome/browser/profiles/profile_impl.cc:1103:3
  01c22671  ProfileImpl::OnPrefsLoaded(Profile::CreateMode, bool)                             ../../../src/chrome/browser/profiles/profile_impl.cc:1140:3
```

```
05-29 23:36:37.857: E/chromium(22344): [ERROR:shell_integration_android.cc(22)] Not implemented reached in shell_integration::DefaultWebClientSetPermission shell_integration::GetDefaultWebClientSetPermission()
05-29 23:36:37.887: W/System.err(22344): java.lang.AssertionError: Only one native client can be registered, even if the previous one was unregistered
05-29 23:36:37.888: W/System.err(22344): 	at org.chromium.chrome.browser.sync.TrustedVaultClient.registerNative(TrustedVaultClient.java:221)
05-29 23:36:37.888: W/System.err(22344): 	at android.os.MessageQueue.nativePollOnce(Native Method)
05-29 23:36:37.888: W/System.err(22344): 	at android.os.MessageQueue.next(MessageQueue.java:335)
05-29 23:36:37.888: W/System.err(22344): 	at android.os.Looper.loopOnce(Looper.java:161)
05-29 23:36:37.888: W/System.err(22344): 	at android.os.Looper.loop(Looper.java:288)
05-29 23:36:37.888: W/System.err(22344): 	at android.app.ActivityThread.main(ActivityThread.java:7886)
05-29 23:36:37.888: W/System.err(22344): 	at java.lang.reflect.Method.invoke(Native Method)
05-29 23:36:37.888: W/System.err(22344): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
05-29 23:36:37.888: W/System.err(22344): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
```
We did the above together with @AlexeyBarabash, but bringing him in on whether we need `TrustedVaultClientAndroid`.

Reply:
I conform that approach at https://github.com/brave/brave-core/pull/13757/commits/0095e2b03ab0bca788868438db545f63a0ee3c2f works for Android

Reply:
Thanks @AlexeyBarabash!

Reply:
Removed the patch, introduced `NoOpTrustedVaultClient` and overrode `TrustedVaultClientAndroid` to be `NoOpTrustedVaultClient` in `chrome_sync_client.cc`.